modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/MergeJoinNode.java

```diff
@@ -962,10 +965,13 @@ else if (cmp > 0) {
                     }
                     else {
                         if (left == null) {
-                            if (waitingLeft == NOT_WAITING)
+                            if (waitingLeft == NOT_WAITING) {
+                                rightIdx = 0;
                                 rightMaterialization = null;
+                                drainMaterialization = false;
```

Comment:
Should we introduce a methods like
```
        /** */
        private void clearRightMaterialization() {
            rightMaterialization = null;
            drainMaterialization = false;
            rightIdx = 0;
        }
```
?
Sometimes we do
```
   rightIdx = 0;
   rightMaterialization = null;
```
without
`drainMaterialization = false`

Reply:
We do this without `drainMaterialization = false` only under the branch `if (!drainMaterialization)`, so `drainMaterialization = false` is useless in this case.